File: challenges/3-area-trapezio/js/caiomartins1/area-trapezio.js
// Esta função certifica-se que todos os valores recebidos são numeros, e retorna o resultado calculado pela formula.
function areaTrapezio(baseMenor, baseMaior, altura) {
  if (isNaN(baseMenor) || isNaN(baseMaior) || isNaN(altura)) {
    console.log('Valores Invalidos!');
    return;
  }
  return ((baseMaior + baseMenor) * altura) / 2;
}

// Faz o cast para Number do input recebido por linha de comandos e salva na variavel input.
const input = process.argv.slice(2).map((el) => Number(el));

// Caso o usuario não dê o numero de inputs esperado.
if (input.length !== 3) {
  console.log('Numero de argumentos invalido!');
  console.log('Uso correto: node area-trapezio.js baseMenor baseMaior altura');
  return;
}

const area = areaTrapezio(...input);

if (!area) return;

console.log('A área do trapézio é: ' + areaTrapezio(...input));

File: challenges/3-area-trapezio/java/wesleyrcb/AreaTrapezio.java
import java.util.Scanner;

public class AreaTrapezio{
	public static void main(String[] args){
		Scanner scan = new Scanner(System.in);
		double baseMaior, baseMenor, altura, area;

		System.out.print("baseMaior: ");
		baseMaior = scan.nextDouble();
		System.out.print("baseMenor: ");
		baseMenor = scan.nextDouble();
		System.out.print("altura: ");
		altura = scan.nextDouble();

		area = (baseMaior + baseMenor) * altura / 2;

		System.out.printf("A área do trapézio é: %.1f", area);

	}
}